### PR TITLE
Add a performance section to TPU docs to address FAQ 

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Here is me, a total noob when it comes to TPU, trying to summarize performance considerations when training pytorch code on TPU. 

Need a bit of a fact check. 
Please edit this draft directly as you please, any improvements welcome!

### CODE REVIEW

**Review by lezwon — APPROVED** [review]

File: docs/source/tpu.rst [hunk 1]
```diff
@@ -194,6 +193,28 @@ Under the hood the xla library will use the `bfloat16 type <https://en.wikipedia
...
+Performance considerations
+--------------------------
+
+The TPU was designed for specific workloads and operations to carry out large volumes of matrix multiplication,
+convolution operations and other commonly used ops in applied deep learning.
+The specialization makes it a strong choice for NLP tasks, sequential convolutional networks, and under low precision operation.
+There are cases in which training on TPUs is slower when compared with GPUs, for possible reasons listed:
+
+- Too small batch size.
+- Explicit evaluation of tensors during training. e.g `tensor.item()`
```
Comment (rohitgr7) [inline docs/source/tpu.rst:205]:
```suggestion
- Explicit evaluation of tensors during training. e.g ``tensor.item()``
```

**Review by rohitgr7 — APPROVED** [review]
include this too? https://github.com/pytorch/xla/blob/master/TROUBLESHOOTING.md#xla-tensor-quirks

some info about weight sharing when using a model like Albert with TPU. @lezwon is working on a fix for it.

**Comment by lezwon** [comment]
@rohitgr7 #5541 addresses that in the docs.

**Review by tchaton — APPROVED** [review]
Great addition !

**Review by justusschock — APPROVED** [review]

**Pull request merged**